provider/core/src/struct_provider.rs

```diff
@@ -5,54 +5,47 @@
 //! Data provider always serving the same struct.
 
 use crate::prelude::*;
-use crate::yoke::trait_hack::YokeTraitHack;
-use crate::yoke::*;
 
-/// A data provider that returns clones of a constant data payload.
+/// A data provider that returns clones of a constant type-erased payload.
 ///
 /// # Examples
 ///
 /// ```
 /// use icu_provider::prelude::*;
 /// use icu_provider::hello_world::*;
-/// use icu_provider::struct_provider::StructProvider;
+/// use icu_provider::struct_provider::AnyPayloadProvider;
 /// use std::borrow::Cow;
 ///
-/// let local_data = HelloWorldV1 {
-///     message: Cow::Owned("hello world".to_string()),
+/// const CONST_DATA: HelloWorldV1<'static> = HelloWorldV1 {
+///     message: Cow::Borrowed("hello world"),
 /// };
 ///
 /// // A placeholder key to use to serve the data struct
 /// const SAMPLE_KEY: ResourceKey = icu_provider::resource_key!(x, "xyz", "example", 1);
 ///
-/// let provider = StructProvider {
+/// let provider = AnyPayloadProvider {
 ///     key: SAMPLE_KEY,
-///     data: DataPayload::from_owned(local_data),
+///     data: AnyPayload::from_static_ref(&CONST_DATA),
 /// };
 ///
-/// let payload: DataPayload<HelloWorldV1Marker> = provider.load_payload(&DataRequest::from(SAMPLE_KEY))
+/// let payload: DataPayload<HelloWorldV1Marker> = provider.load_any(&DataRequest::from(SAMPLE_KEY))
 ///     .expect("Load should succeed")
 ///     .take_payload()
-///     .expect("Data should be present");
+///     .expect("Data should be present")
+///     .downcast()
+///     .expect("Types should match");
 ///
 /// assert_eq!(payload.get().message, "hello world");
 /// ```
-pub struct StructProvider<M>
-where
-    M: DataMarker,
-{
+pub struct AnyPayloadProvider {
     pub key: ResourceKey,
-    pub data: DataPayload<M>,
+    pub data: AnyPayload,
```

Comment:
thought(nb): These fields being public are fine but it might be nice API-wise to have a convenient constructor

Reply:
This is a pre-existing issue, and I may make further revisions to this class after I do the other issues I have lined up, so I would rather not add a constructor right now.